grommet/hpe-design-system
# Updated DataTable vertical list example table

[Deploy Preview](https://deploy-preview-2705--keen-mayer-a86c8b.netlify.app/components/datatable#vertically-align-row-data)

#### What does this PR do?
Creates a DataTable example in design-system.hpe.design/components/datatable to show functionality of a DataTable with a vertical list in data row entries. Provides guidelines on using, formatting, etc. for DataTables that may contain vertical lists.

#### Where should the reviewer start?
/src/examples/components/datatable/DataTableVerticalListExample.js
/src/data/mockData/serverGroups.json
/src/pages/components/datatable.mdx

#### What testing has been done on this PR?
Manually tested functionality to match spec detailed in issue #2455. 

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**

**Accessibility Checks**

**Code Quality Checks**

#### How should this be manually tested?
Make sure that snapshot is reasonable and when the markdown guidance is written/hyperlinks are correctly routed, make sure they route properly and the example code displays properly.

#### Any background context you want to provide?
Previously, there were no guidelines for how a DataTable containing vertical lists should be handled. The "Example Code" tab on the site for the vertical list example will show 404 error until these commits are merged with the main repository.

#### What are the relevant issues?
Issue #2455 

#### Screenshots (if appropriate)
<img width="548" alt="image" src="https://user-images.githubusercontent.com/23144209/177850613-d99030cb-462a-4cff-8025-6c5fe6d10d95.png">
<img width="667" alt="image" src="https://user-images.githubusercontent.com/23144209/177850664-9482842e-1973-4acc-bd90-92b7c7ed4fb1.png">




#### Should this PR be mentioned in Design System updates?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.
